### ISSUE TITLE
[BUG] Multi select field doesn't save values.

### ISSUE THREAD

**Issue by marcingajda** [bolt/bolt#5965]
Multi select fields (those tag-like ones) don't keep values. After saving and refreshing the editor the field is empty. 

Details
-------

 - Relevant Bolt Version: 3.2.0 RC1
 - Install type: Composer install
 - PHP version: 7.0
 - Used web server: Nginx

Reproduction
------------

 1. **Bug summary**: 
    * Saving multi select field doesn't work
    * After save and editor refresh the field is empty
 2. **Specifics**:
    * It happens in the record editor
    * I'm using Bolt 3.2.0 RC1 installed with composer
    * I'm using Nginx, PHP 7.0 and MySQL database
    * Tested in Chrome beta
 3. **Steps to reproduce**:
    * Create a content type with multi select field which use records of another type    
    * Add new record of this type
    * Choose some values in this field
    * Save the record
    * Refresh the page
 4. **Expected result**: 
    * Set values should be shown in the editor after refreshing
 5. **Actual result**: 
    * The multi select field is empty

```
presentation_pl:
    name: Presentations (PL)
    singular_name: Presentation (PL)

    slug: presentations_pl
    singular_slug: presentation_pl
    fields:
        <<: *presentation_fields
        speakers:
            type: select
            multiple: true
            values: speaker_pl/anchor
```

**Comment by bobdenotter** [comment]
I've not been able to reproduce this yet.. Here's what I put in my contenttypes.yml: 

```
presentations_pl:
    name: Presentations (PL)
    singular_name: Presentation (PL)
    slug: presentations_pl
    singular_slug: presentation_pl
    fields:
        title:
            type: text
            class: large
            group: content
        slug:
            type: slug
            uses: title
        speakers:
            type: select
            multiple: true
            values: speaker_pl/anchor

speaker_pl:
    name: Speakers (PL)
    singular_name: Speaker (PL)
    fields:
        title:
            type: text
            class: large
            group: content
        slug:
            type: slug
            uses: title
        anchor:
            type: text
            class: large
        content:
            type: html
            height: 150px
```

Then, update the DB, insert some loripsum. 

Screencast: http://recordit.co/QWnQKQir7R

Can you try it with what I have, to see if it works? If not, can you post the entirety of the presentation_pl and speaker_pl contenttypes?

**Comment by marcingajda** [comment]
Ok, for your code it works perfectly but not for mine. Here is mine full configuration.

```
__nodes:
    speaker_fields: &speaker_fields
        first_name:
            type: text
            class: large
            group: details
            label: First name
        last_name:
            type: text
            class: large
            group: details
            label: Second name
        anchor:
            type: text
            class: small
            group: details
            label: Anchor
            info: >
              Slug which will be use in URL like for
              example example.com/lorem#jankowalski
        company:
            type: text
            label: Company name
        company_url:
            type: text
            label: Companys URL address
        description:
            type: html
            height: 300px
        photo:
            type: image
            label: Photo
    presentation_fields: &presentation_fields
        title:
            type: text
            group: Details
        details:
            type: text
            info: Hour and auditorium
        description:
            type: html
        path:
            type: select
            values:
              path_one 'Path 1'
              path_two: 'Path 2'
              path_three: 'Path 3'


speaker_pl:
    name: Speakers (PL)
    singular_name: Speaker (PL)

    slug: speakers_pl
    singular_slug: speaker_pl
    show_in_menu: 'Resources'
    viewless: true
    title_format: [first_name, last_name]

    fields:
        <<: *speaker_fields

    icon_many: fa:microphone
    icon_one: fa:microphone

presentation_pl:
    name: Presentations (PL)
    singular_name: Presentation (PL)

    slug: presentations_pl
    singular_slug: presentation_pl
    show_in_menu: 'Resources'
    viewless: true
    title_format: title

    fields:
        <<: *presentation_fields
        speakers:
            type: select
            multiple: true
            values: speaker_pl/anchor
    icon_many: fa:file-image-o
    icon_one: fa:file-image-o
```

**Comment by marcingajda** [comment]
Ok, actually the problem is with a different one. The values are saved but badly.

```
mysql> select speakers from bolt_presentations_pl;
+----------+
| speakers |
+----------+
| Array    |
| Array    |
|          |
|          |
+----------+
```

**Comment by bobdenotter** [comment]
Pinging @rossriley Is this something you'd know from the top of your head, or should I dig deeper?

**Comment by rossriley** [comment]
I'm not 100% sure about this, I'll spin up a test tomorrow and see if I can spot the problem.

One thing I do remember is that in the past I've had trouble if I didn't include the `id` in the values although I'm not sure this is related.

For example try this approach:

````
    speakers:
            type: select
            multiple: true
            values: speaker_pl/id,anchor
            keys: anchor
```

**Comment by marcingajda** [comment]
No, it didn't helped. What is interesting the field works after changing `multiple` to `false`.

**Comment by rossriley** [comment]
Cool, leave it with me and I'll do some testing tomorrow.

**Comment by rossriley** [comment]
Ok, the problem seems to manifest itself when the slug doesn't match the contenttype name which is why it's quite subtle, in this case 

```
presentation_pl:
    name: Presentations (PL)
    singular_name: Presentation (PL)
    slug: presentations_pl
```

So if you change the contenttype name to `presentations_pl` then all works ok. I'm guessing that somewhere the contenttype lookup gets confused between using slug or keyname. Fix incoming shortly.

**Comment by bobdenotter** [comment]
Fixed by #5969
